### PR TITLE
Flush after identify or group

### PR DESCRIPTION
**Pull request by vadimdemedes**
Fixes https://github.com/segmentio/analytics-node/issues/95.

As discussed, flushing after `identify()` is necessary in order to preserve traits for future messages. When looking through the docs, I saw that `group()` also has traits, so I added it too.

One question though, should we also flush after `alias()`?

### CODE REVIEW

**Comment by stevenmiller888** [comment]
The end-tools stitch together the anonymous and identified histories irrespective of when the connection is established. Also, we don't guarantee order in the pipeline anyway so I don't think it makes sense for the libraries to handle this.

**Comment by stevenmiller888** [comment]
I can see why this is unexpected for the users of this library though.

**Comment by vadimdemedes** [comment]
In that case, the problem mentioned in #95 shouldn't occur?

> Because we need to be certain that the identify() call completes before calling track(). Otherwise the track() won't carry the trait information sent in identify().

**Comment by stevenmiller888** [comment]
Correct!

**Comment by stevenmiller888** [comment]
It could occur in that events may be delivered out of order. But the end tool should be stitching together the histories so it's okay. Probably worth asking a follow up on why it's important to him that events are delivered in order.

**Comment by vadimdemedes** [comment]
I think it's important for him, because of this:

> Otherwise the track() won't carry the trait information sent in identify().

I'll follow up and confirm that this issue actually happens. Maybe it's a bug on the backend, if it didn't stitch the history correctly.

**Comment by stephenmathieson** [comment]
TLDR; this will not work as expected 

as stated previously, there is no guarantee that events will be processed in the same order the arrive at Segment's "front door" (the tracking API: api.segment.io). there are tons of things that could change the order events are processed. to list a few: an event could get rejected then retried, an NSQ node could explode, a worker could OOM and stop accepting messages while it waits for others to take some load...

because of this, adding workarounds/false promises in the client libraries seems like the wrong move. instead we ought to explain this better in our documentation and make sure success is aware.

**Comment by vadimdemedes** [comment]
Sounds reasonable, thanks @stephenmathieson and @stevenmiller888 for explaining this!
